exercises/outcomes/AT/AT5/generator.sage

```diff
@@ -8,33 +8,59 @@ class Generator(BaseGenerator):
-                    RHS = plus(times(c,v1),times(c,v2))
-                elif prop == "dist_s":
-                    LHS = times(c+d,v)
-                    RHS = plus(times(c,v),times(d,v))
-                LHS = vectorsimplify(LHS)
-                RHS = vectorsimplify(RHS)
-                assert LHS == RHS
-            except AssertionError:
-                raise Exception(f"failed on {prop} {LHS} {RHS}")
-            return vectorsimplify(LHS)
 
+        true_property_options = ["add_assoc","add_comm","mul_assoc","dist_v","dist_s","mul_id"]
+        false_only_property_options = ["add_id","add_inv"]
+
+        def verify(plus,times,hardfalseproperties=[]):
+            trueproperties={}
+            falseproperties=hardfalseproperties
+            for prop in true_property_options:
+                    if prop == "add_assoc":
+                        LHS = plus(v1,plus(v2,v3))
+                        RHS = plus(plus(v1,v2),v3)
+                    elif prop == "add_comm":
+                        LHS = plus(v1,v2)
+                        RHS = plus(v2,v1)
+                    elif prop == "mul_assoc":
+                        LHS = times(c*d,v)
+                        RHS = times(c,times(d,v))
+                    elif prop == "mul_id":
+                        LHS = times(1,v)
+                        RHS = v
+                    elif prop == "dist_v":
+                        LHS = times(c,plus(v1,v2))
+                        RHS = plus(times(c,v1),times(c,v2))
+                    elif prop == "dist_s":
+                        LHS = times(c+d,v)
+                        RHS = plus(times(c,v),times(d,v))
+                    LHS = vectorsimplify(LHS)
+                    RHS = vectorsimplify(RHS)
+                    if LHS == RHS:
+                        trueproperties[prop]=vectorsimplify(LHS)
+                    else:
+                        falseproperties.append(prop)
+            for prop in false_only_property_options:
+                if "dist_s" in trueproperties and "mul_id" in trueproperties:
+                    if prop == "add_id":
+                        LHS = plus(times(0,v),v)
+                        RHS = v
+                        if vectorsimplify(LHS) != vectorsimplify(RHS):
+                            falseproperties.append(prop)
+                        else:
+                            LHS = plus(times(-1,v),v)
+                            RHS = times(0,v)
+                            if vectorsimplify(LHS) != vectorsimplify(RHS):
+                                falseproperties.append("add_inv")
+            return (trueproperties, falseproperties)
+        
+        #Use this to code in false properties that cannot be checked automatically ("add_id" and "add_inv")
+        hardfalseproperties=[]
+
+        #Use this to list a property that is true, but you don't want students to check
+        #because it is too easy (usually "add_comm") or too hard
+        true_no_check_properties=[]
```

Comment:
Sometimes a property is true, but either trivial to check (e.g. addition is commutative), or annoyingly painful.  This allows us to omit those from the list of properties students are asked to check.